pkg/manager/internal.go

```diff
@@ -633,6 +636,7 @@ func (cm *controllerManager) startLeaderElection(ctx context.Context) (err error
 			},
 		},
 		ReleaseOnCancel: cm.leaderElectionReleaseOnCancel,
+		Name:            cm.leaderElectionID,
```

Comment:
why we were not passing the id here before? 
were not it required before as well?
why it is required now and what is the relation of it with the proposed changes?

Reply:
Because the newly enabled clientgo leader election metrics needs the name of the leader election resource, which is the purpose of this parameter.